### PR TITLE
Extract block draggable scroll behaviour into React hook

### PR DESCRIPTION
**Pull request by talldan**
## Description
As mentioned in https://github.com/WordPress/gutenberg/pull/23082/files#r444178859, the new scroll behavior when dragging can be pretty neatly encapsulated into a React hook.

This PR moves the code into a hook, pretty much verbatim, with a few minor bits of renaming and updating of comments to meet coding standards.

## How has this been tested?
- Block dragging/scrolling behavior should be the same as in `master`.

## Types of changes
Non-breaking code quality change

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
I like this refactoring, I also wonder whether we can remove the "key" and the changes to BlockPopover from the other PR too.

File: packages/block-editor/src/components/block-draggable/index.js [hunk 1]
```diff
@@ -65,12 +42,12 @@ const BlockDraggable = ( {
 	);
 	const isDragging = useRef( false );
 
-	// @todo - do this for horizontal scroll
-	const dragStartY = useRef( null );
-	const velocityY = useRef( null );
-	const scrollParentY = useRef( null );
-
-	const scrollEditorInterval = useRef( null );
+	const blockElementId = `block-${ clientIds[ 0 ] }`;
```
[the remark this answers is not in the document]
Comment (talldan) [reply packages/block-editor/src/components/block-draggable/index.js]: Oh yep, I recall seeing that. I've changed the code to use that.

I noticed as well that the code in the hook is fairly agnostic to any kind of drag/drop event, so I went with making the hook accept a `dragElement` and `BlockDraggable` now grabs the block's element from `BlockNodes` context to pass into the hook.

[the remark this answers is not in the document]
Comment (talldan) [reply packages/block-editor/src/components/block-draggable/index.js]: I didn't see any issues when testing manually, but this change seems to be causing some issues with end-to-end tests.

I'll work on fixing it tomorrow.

**Comment by paaljoachim** [comment]
Can the same procedure be used here?
**Try: Don't show a clone when dragging.**
https://github.com/WordPress/gutenberg/pull/23024

**Comment by talldan** [comment]
Tests are now passing, apart from PHP tests which seem to have started failing overnight. I can't see any merged PRs that might have caused it, so it's most likely an environment issue.

Using admin privileges to merge this as it contains no PHP changes.

**Pull request merged**